### PR TITLE
feat: EventChannel

### LINKED ISSUE #103: Add support for EventChannel
https://docs.flutter.io/flutter/services/EventChannel-class.html

### PR DESCRIPTION
**Pull request by pchampio**
fixes #103 
WIP.

Everything is good except for the `EndOfStream` impl https://github.com/flutter/flutter/issues/35434.

### CODE REVIEW

**Comment by Drakirus** [comment]
I'm making progress on the implementation. I had a hard time figuring out what to do when the user triggers a hot restart. 
The implementation survive hot restart, without wasting too mush resources!
<details>

<summary> Here is a example of a Golang EventChannel (Clickable) </summary>

```go
package main

import (
	"time"

	"github.com/go-flutter-desktop/go-flutter"
	"github.com/go-flutter-desktop/go-flutter/plugin"
)

var options = []flutter.Option{
	flutter.PopBehavior(flutter.PopBehaviorClose), // on SystemNavigator.pop() closes the app

	flutter.AddPlugin(&test{}),
}

const textinputChannelName = "com.yourcompany.eventchannelsample/stream"

type test struct {
	stop chan bool
}

var _ flutter.Plugin = &test{}

func (p *test) InitPlugin(messenger plugin.BinaryMessenger) error {
	p.stop = make(chan bool)

	channel := plugin.NewEventChannel(messenger, textinputChannelName, plugin.StandardMethodCodec{})
	// channel := plugin.NewEventChannel(messenger, textinputChannelName, plugin.JSONMethodCodec{})

	channel.Handle(p)
	print(channel)
	return nil
}

func (p *test) OnListen(arguments interface{}, sink *plugin.EventSink) {
	var i int32
	i = 0
	for {
		select {
		case <-p.stop:
			return
		default:
			i++
			time.Sleep(100 * time.Millisecond)
			// print("\nsend\n")
			sink.Success(i)
		}
	}
}
func (p *test) OnCancel(arguments interface{}) {
	// I choose to use channels to Cancel events. 
	// Mutex can also work.
	// I found that channels are bit more reliable than Mutex during hot restart. 
	p.stop <- true
}

```
</details>

  matching the sample found [here](https://github.com/seamusv/event_channel_sample)

File: embedder/embedder.go
Comment (GeertJohan) [inline embedder/embedder.go]: Just checking, is Success the result we should give back when shutting down?

Comment (Drakirus) [reply embedder/embedder.go]: yeaa, we should create another result for this use-case.

Comment (Drakirus) [reply embedder/embedder.go]: Wrapping [`m.engine.SendPlatformMessage(msg)`](https://github.com/go-flutter-desktop/go-flutter/blob/b21af7bbc3c034023d9cf4b86ea765e7097031dc/messenger.go#L41) into a [`tasker.do`](https://github.com/go-flutter-desktop/go-flutter/blob/b21af7bbc3c034023d9cf4b86ea765e7097031dc/messenger.go#L107-L112) could also solves to issue. Not a big fan of  `tasker.do` because it will make replies harder to implement.

**Comment by Drakirus** [comment]
Waiting for a PR to be merged on flutter/engine #9781
The golang implementation is good to be reviewed.

**Comment by Drakirus** [comment]
Waiting for a PR to be merged on flutter/engine #9781
The golang implementation is good to be reviewed.

File: plugin/event-channel.go
Comment (GeertJohan) [inline plugin/event-channel.go]: *stream

File: plugin/event-sink.go
Comment (GeertJohan) [inline plugin/event-sink.go]: Formatting nit, remove empty first lines in function block?

Comment (GeertJohan) [reply plugin/event-sink.go]: Also in Error and EndOfStream

**Review by GeertJohan — CHANGES_REQUESTED** [review]
Just a spelling mistake and some formatting, otherwise looks good and can be merged!

**Comment by Drakirus** [comment]
I'll merge this PR once flutter/engine #9781 gets merged master

**Review by GeertJohan — APPROVED** [review]

**Pull request merged**
